lib/webdriver.py

```diff
@@ -89,6 +91,9 @@ def set_headers(self) -> None:
         self._take_debug_screenshot(driver, "post_headers.png")
 
         driver.quit()
+        if self.display is not None:
```

Comment:
We should move this to a separate function (e.g. `_quit_driver(driver)`) to reduce code repetition in `get_reservations` and `_wait_for_login` as well.